src/Magnum/SceneGraph/CMakeLists.txt

```diff
@@ -0,0 +1,60 @@
+#
+#   This file is part of Magnum.
+#
+#   Copyright © 2010, 2011, 2012, 2013, 2014, 2015
+#             Vladimír Vondruš <mosra@centrum.cz>
+#   Copyright © 2015 Andrea Capobianco <andrea.c.2205@gmail.com>
+#   Copyright © 2015 Jonathan Hale <squareys@googlemail.com>
+#
+#   Permission is hereby granted, free of charge, to any person obtaining a
+#   copy of this software and associated documentation files (the "Software"),
+#   to deal in the Software without restriction, including without limitation
+#   the rights to use, copy, modify, merge, publish, distribute, sublicense,
+#   and/or sell copies of the Software, and to permit persons to whom the
+#   Software is furnished to do so, subject to the following conditions:
+#
+#   The above copyright notice and this permission notice shall be included
+#   in all copies or substantial portions of the Software.
+#
+#   THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+#   IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+#   FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL
+#   THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+#   LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+#   FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
+#   DEALINGS IN THE SOFTWARE.
+#
+
+cmake_minimum_required(VERSION 2.8.12)
+
+find_package(Magnum REQUIRED SceneGraph Shapes)
+
+set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} ${CORRADE_CXX_FLAGS}")
+
+include_directories(${MAGNUM_INCLUDE_DIRS} ${CMAKE_CURRENT_BINARY_DIR})
+
+set(MagnumOctreeDrawableGroup_SOURCES
+    instantiation.cpp)
+
+set(MagnumOctreeDrawableGroup_HEADERS
+    OctreeDrawableGroup.h
+    OctreeDrawableGroup.hpp)
+
+add_library(MagnumOctreeDrawableGroup STATIC ${MagnumOctreeDrawableGroup_HEADERS} ${MagnumOctreeDrawableGroup_SOURCES})
+target_link_libraries(MagnumOctreeDrawableGroup
+    ${MAGNUM_LIBRARIES} ${MAGNUM_SCENEGRAPH_LIBRARIES} ${MAGNUM_SHAPES_LIBRARIES} MagnumOctree)
+
+target_include_directories(MagnumOctreeDrawableGroup PUBLIC
+    ${PROJECT_SOURCE_DIR}/src
+    ${PROJECT_BINARY_DIR}/src
+    $<TARGET_PROPERTY:Magnum::Magnum,INTERFACE_INCLUDE_DIRECTORIES>)
+
+install(TARGETS MagnumOctreeDrawableGroup
+    RUNTIME DESTINATION ${MAGNUM_BINARY_INSTALL_DIR}
+    LIBRARY DESTINATION ${MAGNUM_LIBRARY_INSTALL_DIR}
+    ARCHIVE DESTINATION ${MAGNUM_LIBRARY_INSTALL_DIR})
+install(FILES ${MagnumOctreeDrawableGroup_HEADERS} DESTINATION ${MAGNUM_INCLUDE_INSTALL_DIR}/SceneGraph)
+    
+if(${BUILD_TESTS})
+    add_subdirectory(Test)
+endif()
```

Comment:
Yup, exactly. In other words, the `Magnum/SceneGraph/` directory is "owned" by the SceneGraph library (and basically each directory has a library that owns it) and no other library should put its files into it. This also makes it possible to immediately know which library does which include come from, i.e. if the compiler complains about `Magnum/Octree/OctreeDrawableGroup.h` missing, then you know that you forgot to install / link to the Octree library. If it would complain about `Magnum/SceneGraph/OctreeDrawableGroup.h`, then you would be unsure at best. (Is there a new revision of SceneGraph that contains this header? Is it coming from some other library? Which one, exactly? Did I install SceneGraph library wrong? Is this a typo? Why is everything so terrible?)

*(Side note: I yet have to go through all theother comments, replied just to this ATM.)*